dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java

```diff
@@ -443,23 +452,31 @@ public void run() {
                 lastChecked = current;
                 return;
             }
+
             // find available processors
             int runtimeAvailableProcessors = Runtime.getRuntime().availableProcessors();
-            int fileSystemAvailableProcessors = getAvailableProcessorsFromFilesystem();
+            float fileSystemAvailableProcessors = getAvailableProcessorsFromFilesystemFloat();
 
-            int newAvailableProcessors;
+            int newAvailableProcessorsInt;
+            int newAvailableProcessorsFloat;
             if (fileSystemAvailableProcessors <= 0 || fileSystemAvailableProcessors > runtimeAvailableProcessors) {
-                newAvailableProcessors = runtimeAvailableProcessors;
+                newAvailableProcessorsInt = runtimeAvailableProcessors;
+                newAvailableProcessorsFloat = runtimeAvailableProcessors * 100;
             } else {
-                newAvailableProcessors = fileSystemAvailableProcessors;
+                newAvailableProcessorsInt = roundUpToNextInt(fileSystemAvailableProcessors);
+                newAvailableProcessorsFloat = (int) (fileSystemAvailableProcessors * 100);
             }
 
-            int currentNumberOfProcessors = AVAILABLE_PROCESSORS.get();
-            if (currentNumberOfProcessors != newAvailableProcessors) {
-                if (AVAILABLE_PROCESSORS.compareAndSet(currentNumberOfProcessors, newAvailableProcessors)) {
-                    notifyListeners(newAvailableProcessors);
+            int currentNumberOfProcessorsInt = AVAILABLE_PROCESSORS_INTEGER.get();
+            int currentNumberOfProcessorsFloat = AVAILABLE_PROCESSORS_FLOAT.get();
+
+            if (currentNumberOfProcessorsInt != newAvailableProcessorsInt || currentNumberOfProcessorsFloat != newAvailableProcessorsFloat) {
+                AVAILABLE_PROCESSORS_FLOAT.set(newAvailableProcessorsFloat);
+                if (AVAILABLE_PROCESSORS_INTEGER.compareAndSet(currentNumberOfProcessorsInt, newAvailableProcessorsInt)) {
+                    notifyListeners(newAvailableProcessorsInt);
```

Comment:
```suggestion
            if (currentNumberOfProcessorsFloat != newAvailableProcessorsFloat) {
                AVAILABLE_PROCESSORS_FLOAT.set(newAvailableProcessorsFloat);
            }
            if (currentNumberOfProcessorsInt != newAvailableProcessorsInt) {
                if (AVAILABLE_PROCESSORS_INTEGER.compareAndSet(currentNumberOfProcessorsInt, newAvailableProcessorsInt)) {
                    notifyListeners(newAvailableProcessorsInt);
```

I thought about this during my lunch hour and I think this is cleaner and more accurate.